frontend/client/templates/offermodal.js

```diff
@@ -1,20 +1,99 @@
-Template.offermodal.helpers({
-  total: function (price, volume) {
+Template.offermodal.viewmodel({
+  volume: '',
+  total: '',
+  autorun: function () {
+    if (Template.currentData().offer) {
+      var volume = web3.fromWei(new BigNumber(Template.currentData().offer.volume))
+      var price = web3.fromWei(new BigNumber(Template.currentData().offer.price))
+      var total = volume.times(price)
+      this.volume(volume.toString(10))
+      this.total(total.toString(10))
+    }
+  },
+  maxVolume: function () {
     try {
-      return web3.fromWei(new BigNumber(price)).times(new BigNumber(volume)).toString(10)
+      if (Template.currentData().offer.type === 'bid') {
+        // Calculate max volume, since we want to sell MKR, we need to check how much MKR we can sell
+        var token = Tokens.findOne('MKR')
```

Comment:
Not sure you meant to hardcode MKR here - shouldn't it be `Template.currentData().offer.currency`?